.dockerignore

```diff
@@ -1,8 +1,29 @@
-.vscode
-.circleci
+# Hidden files and directories.
+.*
 
-vendor/
+# Top-level directories unrelated to the build.
+docs/
 infrastructure/
+scripts/
+staker-rewards/
 
-solidity/node_modules/
-solidity/build/
+# Top-level files unrealted to the build.
+CODEOWNERS
+crytic-config.json
+Dockerfile
+*.adoc
+
+# NPM stuff.
+**/node_modules/*
+
+# Solidity stuff.
+# We want to include only bare contracts and NPM package configuration for Go code
+# generator.
+solidity/
+!solidity/contracts
+!solidity/package.json
+!solidity/package-lock.json
```

Comment:
How much does this gain? It feels like most of the extra heft from the solidity dir should be hit by the node_modules exclusion above, is that not the case?

Reply:
Yep, you're right. Maybe I was trying to make it too kosher. From the CI perspective, there are mostly files that are not heavy but unnecessary in the image. From the local env perspective, there's also `build` directory which is ~37MB. The filters are not about size but more about purity.

```
✗ du -hs solidity/* | sort -rh | head -10
715M    solidity/node_modules
 37M    solidity/build
1.1M    solidity/package-lock.json
412K    solidity/test
328K    solidity/contracts
 60K    solidity/scripts
 24K    solidity/migrations
 16K    solidity/integration
4.0K    solidity/truffle.js
4.0K    solidity/test-environment.config.js
```